layers/+lang/c-c++/packages.el

```diff
@@ -41,23 +41,27 @@
     srefactor
     xcscope
     ;; lsp
-    (ccls :requires lsp-mode)
+    (ccls :requires lsp-mode :toggle (eq c-c++-backend 'lsp-ccls))
     dap-mode
     ;; rtags
-    (company-rtags :requires (company rtags))
-    (flycheck-rtags :requires (flycheck rtags))
+    (company-rtags :requires (company rtags) :toggle (eq c-c++-backend 'rtags))
+    (flycheck-rtags :requires (flycheck rtags) :toggle (eq c-c++-backend 'rtags))
     ggtags
-    (helm-rtags :requires (helm rtags))
-    (ivy-rtags :requires (ivy rtags))
-    rtags
+    (helm-rtags :requires (helm rtags)
+                :toggle (and (eq c-c++-backend 'rtags)
+                             (configuration-layer/layer-used-p 'helm)))
+    (ivy-rtags :requires (ivy rtags)
+               :toggle (and (eq c-c++-backend 'rtags)
+                            (configuration-layer/layer-used-p 'ivy)))
```

Comment:
I think the `:requires ivy` is probably sufficient, no need for `layer-used-p` here IMO.  (Similarly for helm)